test/e2e/storage/persistent_volumes-local.go

```diff
@@ -252,6 +253,88 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 		})
 	})
 
+	Context("when pod using local volume with non-existant path", func() {
+		ep := &eventPatterns{
+			reason:  "FailedMount",
+			pattern: make([]string, 2)}
+		ep.pattern = append(ep.pattern, "MountVolume.SetUp failed")
+		ep.pattern = append(ep.pattern, "does not exist")
+
+		It("should not be able to mount", func() {
+			for _, testVolType := range LocalVolumeTypes {
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol := &localTestVolume{
+					node:            config.node0,
+					hostDir:         "/non-existent/location/nowhere",
+					localVolumeType: testVolType,
+				}
+				By("Creating local PVC and PV")
+				createLocalPVCPV(config, testVol)
+				pod, _ := createLocalPod(config, testVol)
+				checkPodEvents(config, pod.Name, ep)
+			}
+		})
+	})
+
+	Context("when pod has NodeSelector different from PV's node affinity", func() {
+		ep := &eventPatterns{
+			reason:  "FailedScheduling",
+			pattern: make([]string, 2)}
+		ep.pattern = append(ep.pattern, "MatchNodeSelector")
+		ep.pattern = append(ep.pattern, "No nodes are available")
+
+		It("should not be able to mount due to different affinity", func() {
+			if len(config.nodes.Items) < 2 {
+				framework.Skipf("Runs only when number of nodes >= 2")
+			}
+			for _, testVolType := range LocalVolumeTypes {
```

Comment:
Can the for loop go outside the It() so that you don't have to repeat this for every test case?  If you can move it outside, then the setupLocalVolume can go into a BeforeEach() as well.

Reply:
done